src/protocols/multisig/mod.rs

```diff
@@ -0,0 +1,154 @@
+//!
+//See (https://pdfs.semanticscholar.org/6bf4/f9450e7a8e31c106a8670b961de4735589cf.pdf)
+use cryptography_utils::{BigInt, FE, GE, PK, SK};
+use cryptography_utils::cryptographic_primitives::proofs::*;
+use cryptography_utils::elliptic::curves::traits::*;
+
+use cryptography_utils::cryptographic_primitives::hashing::hash_sha256::HSha256;
+use cryptography_utils::cryptographic_primitives::hashing::traits::*;
+
+use cryptography_utils::arithmetic::traits::Converter;
+use cryptography_utils::arithmetic::traits::Modulo;
+use cryptography_utils::cryptographic_primitives::commitments::hash_commitment::HashCommitment;
+use cryptography_utils::cryptographic_primitives::commitments::traits::*;
+use protocols::multisig;
+#[derive(Debug, Clone)]
+pub struct Keys {
+    pub I: KeyPair,
+    pub X: KeyPair,
+}
+
+#[derive(Debug, Clone)]
+pub struct KeyPair {
+    pub public_key: GE,
+    private_key: FE,
+}
+
+impl KeyPair {
+    pub fn create() -> KeyPair {
+        let ec_point: GE = ECPoint::generator();
+        let private_key: FE = ECScalar::new_random();
+        let public_key = ec_point.scalar_mul(&private_key.get_element());
+        KeyPair {
+            public_key,
+            private_key,
+        }
+    }
+
+    pub fn create_from_private_key(private_key: &BigInt) -> KeyPair {
+        let ec_point: GE = ECPoint::generator();
+        let private_key: FE = ECScalar::from(private_key);
+        let public_key = ec_point.scalar_mul(&private_key.get_element());
+        KeyPair {
+            public_key,
+            private_key,
+        }
+    }
+}
+
+impl Keys {
+    //TODO:: add create from private key
+    pub fn create() -> Keys{
+        let I = KeyPair::create();
+        let X = KeyPair::create();
+        Keys{
+            I,
+            X
+        }
+    }
+
+    pub fn broadcast( keys: &Keys) ->Vec<&GE>{
+       return vec![&keys.I.public_key,&keys.X.public_key];
+    }
+
+    pub fn collect_and_compute_challenge(ix_vec: &Vec<Vec<&GE>>) -> FE{
+        let mut new_vec: Vec<&GE>  = Vec::new();
+        let concat_vec = ix_vec.iter()
+            .fold(new_vec, |mut acc, mut x| {acc.extend(x); return acc});
+        let e = multisig::hash_4(&concat_vec);
+        e
+    }
+
+}
+
+pub fn partial_sign(keys: &Keys, e: &FE) -> FE{
+    let es = e.mul(&keys.I.private_key.get_element());
+    let y = es.add(&keys.X.private_key.get_element());
+    return y;
+}
+
+pub fn verify(I: &GE, X: &GE, y: &FE, e: &FE) ->Result<(),()>{
+    let base_point: GE = ECPoint::generator();
+    let yG = base_point.scalar_mul(&y.get_element());
+    let I_c = I.clone();
+    let eI =  I_c.scalar_mul(&e.get_element());
+    let X_plus_eI = X.add_point(&eI.get_element());
+    if yG.get_element()==X_plus_eI.get_element() {Ok(()) } else{Err(())}
+
+}
+
+pub fn hash_4(key_list: &[&GE]) -> FE{
+    let four_fe: FE = ECScalar::from(&BigInt::from(4));
+    let base_point: GE = ECPoint::generator();
+    let four_ge = base_point * four_fe;
```

Comment:
I actually prefer the multiplication operator for (scalar * ec_point) operations.
Is it possible to switch to using it (instead of `scalar_mul()` method)?